### ISSUE TITLE
Use existing keystore standard or new one

### ISSUE THREAD

**Issue by CarlBeek** [CarlBeek/eth2.0-deposit-tooling#1]
Relevant XKCD:
![The real reason I want to do this](https://imgs.xkcd.com/comics/standards.png)

It is worth considering whether or not to use the [existing (v3) Ethereum standard](https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition) for keystores vs creating a new one.

## Pros for a new standard:
- **Remove the dependency on Keccak:** v3 uses Keccak for checksumming the password. This is drags another hash function into the dependencies of Eth2. (Keccak lives longer in Ethereum)
- **Removal of AES:** v3 uses AES128 in counter mode which is an over-complicated way of mixing in the derived key to obtain the secret (BLS privkey). XOR can achieve the same mixing without introducing an extra crypto assumption (AES assumption) and crypto library dependency. (As far as I can see, AES only exists in keystores due it's historical use in the original [Ethereum-sale keystore generation](https://github.com/ethereum/pyethsaletool/blob/master/pyethsaletool.py#L127))
- **SHA256 - the only assumption:** By making the above changes, SHA256 would be the only assumption required for the security of the system (assuming the use of Scrypt as a KDF (BTW,  the [Scyrpt paper](https://www.tarsnap.com/scrypt/scrypt.pdf) its a fun and witty read if you're feeling bored.) That said, Scrypt's difficulty advantage is rapidly being reduced on account of Script-mining ASICs.)

## Pros for existing standard
- Not introducing a new standard
- Ability to use existing test vectors

**Comment by CarlBeek** [comment]
I'm happy to use the Eth1 Keystore format for now.

What I don't like about it:
* Big Endian encoding (although so is the rest of the Key derivation stack)
* Keccak checksum (It may in fact be immortal.)
* Needlessly complicated (AES etc.)
* Supports both scrypt and PBKDF2 (I would like to push for only scrypt in Eth2. (This is very doable.))

What I do like about it:
* Heaps of existing tooling
* People already understand it

Basically, it serves all of our needs.
